ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/rcf/FixedInTimeRandomCutForest.java

```diff
@@ -0,0 +1,191 @@
+    }
+
+    @Override
+    public Model train(DataFrame dataFrame) {
+        ThresholdedRandomCutForest forest = createThresholdedRandomCutForest(dataFrame);
+        process(dataFrame, forest);
+        Model model = new Model();
+        model.setName(FunctionName.FIT_RCF.name());
+        model.setVersion(1);
+        ThresholdedRandomCutForestState state = trcfMapper.toState(forest);
+         model.setContent(ModelSerDeSer.serialize(state));
+        return model;
+    }
+
+    @Override
+    public MLOutput trainAndPredict(DataFrame dataFrame) {
+        ThresholdedRandomCutForest forest = createThresholdedRandomCutForest(dataFrame);
+        List<Map<String, Object>> predictResult = process(dataFrame, forest);
+        return MLPredictionOutput.builder().predictionResult(DataFrameBuilder.load(predictResult)).build();
+    }
+
+    private List<Map<String, Object>> process(DataFrame dataFrame, ThresholdedRandomCutForest forest) {
+        List<Double> pointList = new ArrayList<>();
+        ColumnMeta[] columnMetas = dataFrame.columnMetas();
+        List<Map<String, Object>> predictResult = new ArrayList<>();
+        for (int rowNum = 0; rowNum< dataFrame.size(); rowNum++) {
+            Row row = dataFrame.getRow(rowNum);
+            long timestamp = -1;
+            for (int i = 0; i < columnMetas.length; i++) {
+                ColumnMeta columnMeta = columnMetas[i];
+                ColumnValue value = row.getValue(i);
+
+                // TODO: sort dataframe by time field with asc order. Currently consider the date already sorted by time.
+                if (timeField != null && timeField.equals(columnMeta.getName())) {
+                    ColumnType columnType = columnMeta.getColumnType();
+                    if (columnType == ColumnType.LONG ) {
+                        timestamp = value.longValue();
+                    } else if (columnType == ColumnType.STRING) {
+                        try {
+                            timestamp = simpleDateFormat.parse(value.stringValue()).getTime();
+                        } catch (ParseException e) {
+                            log.error("Failed to parse timestamp " + value.stringValue(), e);
+                            throw new MLValidationException("Failed to parse timestamp " + value.stringValue());
+                        }
+                    } else  {
+                        throw new MLValidationException("Wrong data type of time field. Should use LONG or STRING, but got " + columnType);
+                    }
+                } else {
+                    pointList.add(value.doubleValue());
+                }
+            }
+            double[] point = pointList.stream().mapToDouble(d -> d).toArray();
+            pointList.clear();
+            Map<String, Object> result = new HashMap<>();
+
+            AnomalyDescriptor process = forest.process(point, timestamp);
```

Comment:
process includes both score and update.  How do you prevent training from happening?

Reply:
Thanks Kaituo, I have the same question and waiting for ML team's response. I created an issue to make this discussion open on RCF repo https://github.com/aws/random-cut-forest-by-aws/issues/299